bin/validate_bigquery

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+import click
+from pathlib import Path
+from git import Repo
+from shutil import copyfile
+import json
+import difflib
+import sys
+
+BASE_DIR = Path("/app").resolve()
+
+
+@click.group()
+def validate():
+    pass
+
+
+def compute_compact_columns(document):
+    def traverse(prefix, columns):
+        res = []
+        for node in columns:
+            name = node["name"] + (".[]" if node["mode"] == "REPEATED" else "")
+            dtype = node["type"]
+            if dtype == "RECORD":
+                res += traverse(f"{prefix}.{name}", node["fields"])
+            else:
+                res += [f"{prefix}.{name} {dtype}"]
+        return res
+
+    res = traverse("root", document)
+    return sorted(res)
+
+
+@validate.command()
+@click.option("--head", default="HEAD")
+@click.option(
+    "--repository",
+    type=click.Path(exists=True, file_okay=False),
+    default=BASE_DIR / "mozilla-pipeline-schemas",
+)
+@click.option(
+    "--artifact",
+    type=click.Path(file_okay=False),
+    default=BASE_DIR / "validate_schema_evolution",
+)
+def copy(head, repository, artifact):
```

Comment:
We could drop this function entirely and use something [like this](https://github.com/mozilla/probe-scraper/blob/master/probe_scraper/scrapers/git_scraper.py#L53) to instead get the files at both commits during `validate`. That would simplify the above bash script and make this a little less reliant on local state.

Reply:
I'm not clear how you would check for whether the file was removed or added between commits. I like the idea of local state outside of version control though, since it makes things easier to inspect.